packages/components/input/src/use-input.ts

```diff
@@ -250,22 +250,34 @@ export function useInput<T extends HTMLInputElement | HTMLTextAreaElement = HTML
   const hasPlaceholder = !!props.placeholder;
   const hasLabel = !!label;
   const hasHelper = !!description || !!errorMessage;
-  const shouldLabelBeOutside = labelPlacement === "outside" || labelPlacement === "outside-left";
+  const shouldLabelBeOutside =
+    labelPlacement === "outside" ||
+    labelPlacement === "outside-left" ||
+    labelPlacement === "outside-top";
   const shouldLabelBeInside = labelPlacement === "inside";
   const isPlaceholderShown = domRef.current
     ? (!domRef.current.value || domRef.current.value === "" || !inputValue || inputValue === "") &&
       hasPlaceholder
     : false;
+
   const isOutsideLeft = labelPlacement === "outside-left";
+  const isOutsideTop = labelPlacement === "outside-top";
 
   const hasStartContent = !!startContent;
+
+  /*
+  outside -> label is outside only when some placeholder is there
+  outside-top, outside-left-> label is outside regardless of placeholder
+*/
   const isLabelOutside = shouldLabelBeOutside
     ? labelPlacement === "outside-left" ||
+      labelPlacement === "outside-top" ||
       hasPlaceholder ||
       (labelPlacement === "outside" && hasStartContent)
     : false;
-  const isLabelOutsideAsPlaceholder =
-    labelPlacement === "outside" && !hasPlaceholder && !hasStartContent;
+  const isLabelOutsideAsPlaceholder = labelPlacement === "outside" && !hasPlaceholder;
+  // &&
+  // !hasStartContent;
```

Comment:
please explain why `!hasStartContent` is removed here

Reply:
I had uncommented this part while making the commit, but it seeems that code was not saved in VS code.
Will remove the comments.